test/interpolating/optional-paths.test.js

```diff
@@ -0,0 +1,34 @@
+var test  = require('tape')
+  , rhumb = require('../../src/rhumb')
+
+test('Interpolating should treat fixed parts are present', function(t){
+  t.plan(1)
+
+  t.equal(rhumb.interpolate('/one/two(/three)', {}), '/one/two/three', 'fixed parts are included in url')
+})
+
+test('Interpolating should process with nested optional fixed parts', function(t){
+  t.plan(2)
+
+  t.equal(rhumb.interpolate('/one/two(/three/four(/five))', {}), '/one/two/three/four/five', 'nested optional elements are included in url')
+  t.equal(rhumb.interpolate('/one/two(/three(/four(/five)))', {}), '/one/two/three/four/five', 'nested optional elements are included in url')
+})
+
+test('Interpolating should process with nested optional variable parts', function(t) {
+  t.plan(5)
+
+  t.equal(rhumb.interpolate('/foo(/{bar}(/{bay}))', { bar: 'knew', bay: 'you' }), '/foo/knew/you', 'all optional parts included when all param present')
+  t.equal(rhumb.interpolate('/foo(/{bar}(/{bay}))', {}), '/foo', 'no optional parts included when no params present')
+  t.equal(rhumb.interpolate('/foo(/{bar}(/{bay}))', { bar: '' }), '/foo', 'no optional parts included when all params are missing or empty')
+  t.equal(rhumb.interpolate('/foo(/{bar}(/{bay}))', { bar: 'knew' }), '/foo/knew', 'include optional parts where params present')
+  t.equal(rhumb.interpolate('/foo(/{bar}(/{bay}))', { bay: 'you' }), '/foo', 'exclude optional parts where where parent optional parts are empty')
```

Comment:
I think that if a optional part is empty, then it should not be included in the URI.